mono/metadata/threads.c

```diff
@@ -2545,14 +2553,11 @@ ves_icall_System_Threading_Thread_ResetAbort (MonoThreadObjectHandle this_obj, M
 	UNLOCK_THREAD (thread);
 
 	if (!was_aborting) {
-		const char *msg = "Unable to reset abort because no abort was requested";
-		mono_set_pending_exception_handle (mono_exception_new_thread_state (msg, error));
-		mono_error_assert_ok (error);
-		return;
-	} else if (is_domain_abort) {
-		/* Silently ignore abort resets in unloading appdomains */
+		mono_error_set_exception_thread_state (error, "Unable to reset abort because no abort was requested");
 		return;
 	}
+	if (is_domain_abort) // Silently ignore abort resets in unloading appdomains
```

Comment:
I think it was clearer as an `else if`.

Reply:
return does not continue through the if ladder. else is also confusing.